### PR TITLE
sos monitoring script

### PR DESCRIPTION
**Pull request by gkaratha**
Script to monitor the HLT_DoubleMu3_DZ_PFMET50_PFMHT60. This path is used by the soft opposite sign analysis for SUSY searches with small mass difference. It uses the SOSMonitor to measure efficiency of both legs (MET and muon)

### CODE REVIEW

File: DQMOffline/Trigger/plugins/SOSMonitor.cc [hunk 1]
```diff
@@ -0,0 +1,316 @@
+#include "DQMOffline/Trigger/plugins/SOSMonitor.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DQM/TrackingMonitor/interface/GetLumi.h"
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/SOSMonitor.cc]: drop it

File: DQMOffline/Trigger/plugins/SOSMonitor.cc [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+#include "CommonTools/TriggerUtils/interface/GenericTriggerEventFlag.h"
+
+#include "TLorentzVector.h"
+
+double sosMAX_PHI = 3.2;
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/SOSMonitor.cc]: add them to the class as members

File: DQMOffline/Trigger/plugins/SOSMonitor.cc [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+  , sos_variable_binning_ ( iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<std::vector<double> >("sosBinning") )
+  , sos_binning_          ( getHistoPSet   (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>   ("sosPSet")    ) )
+  , ls_binning_           ( getHistoLSPSet (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>   ("lsPSet")     ) )
+  , num_genTriggerEventFlag_(new GenericTriggerEventFlag(iConfig.getParameter<edm::ParameterSet>("numGenericTriggerEventPSet"),consumesCollector(), *this))
+  , den_genTriggerEventFlag_(new GenericTriggerEventFlag(iConfig.getParameter<edm::ParameterSet>("denGenericTriggerEventPSet"),consumesCollector(), *this))
+  , metSelection_ ( iConfig.getParameter<std::string>("metSelection") )
+  , jetSelection_ ( iConfig.getParameter<std::string>("jetSelection") )
+  , eleSelection_ ( iConfig.getParameter<std::string>("eleSelection") )
+  , muoSelection_ ( iConfig.getParameter<std::string>("muoSelection") )
+  , njets_      ( iConfig.getParameter<int>("njets" )      )
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/SOSMonitor.cc]: change in unsigned

File: DQMOffline/Trigger/plugins/SOSMonitor.h [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+  MON sosMON_;
+  MON sosMON_variableBinning_;
+  MON sosVsLS_;
+  MON sosPhiEtaMON_;
+  MON sosPhiMON_;
+  MON sosHTMON_;
+
+
+  GenericTriggerEventFlag* num_genTriggerEventFlag_;
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/SOSMonitor.h]: unique_ptr

File: DQMOffline/Trigger/plugins/SOSMonitor.cc [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+  , nelectrons_ ( iConfig.getParameter<int>("nelectrons" ) )
+  , nmuons_     ( iConfig.getParameter<int>("nmuons" )     )
+  , turn_on_    (iConfig.getParameter<std::string>("turn_on"))
+  , met_pt_cut_ (iConfig.getParameter<int>("met_pt_cut"))
+  , mu2_pt_cut_ (iConfig.getParameter<int>("mu2_pt_cut"))
+{
+
+  sosMON_.numerator   = nullptr;
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/SOSMonitor.cc]: do you really need it ?
otherwise, drop it

File: DQMOffline/Trigger/plugins/SOSMonitor.cc [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+
+  edm::Handle<reco::PFMETCollection> metHandle;
+  iEvent.getByToken( metToken_, metHandle );
+  reco::PFMET pfmet = metHandle->front();
+  if ( ! metSelection_( pfmet ) ) return;
+  
+  float met = pfmet.pt();
+  float phi = pfmet.phi();
+  int jet60=0;
```
Comment (mtosi) [inline DQMOffline/Trigger/plugins/SOSMonitor.cc]: ok, I see that all comments I made in your branch have not been addressed yet
please, commits the needed changes
then, I'll continue the review

**Comment by mtosi** [comment]
tracked by https://github.com/cms-sw/cmssw/issues/19142

**Comment by rcastello** [comment]
In case you missed, a new PR has been circulated at https://github.com/cms-sw/cmssw/pull/19337 including the requested changes. Thanks

**Comment by gkaratha** [comment]
Dear Mia, we addressed your comments here: https://github.com/cms-sw/cmssw/pull/19337 . I am not 100% sure if everyone is informed automatically so i posted here as well. Thanks

**Comment by mtosi** [comment]
ciao
i'm in holiday
i've just added some comments in the other PR
if this is no longer needed, please close

**Comment by dmitrijus** [comment]
-1 superseeded

**Comment by fwyzard** [comment]
superseded by which PR ?

**Pull request closed without merging**